### PR TITLE
ToxicChatClassification task added to multilingual

### PR DESCRIPTION
**Pull request by dokato**
## Checklist for adding MMTEB dataset

Reason for dataset addition:

- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] I have considered the size of the dataset and reduced it if it is too big (2048 examples is typically large enough for most tasks)
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [ ] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

### CODE REVIEW

File: mteb/tasks/Classification/multilingual/ToxicChatClassification.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        eval_langs=_LANGS,
+        main_score="accuracy",
+        date=("2023-10-26", "2024-01-31"),
+        form=["written"],
+        domains=["Constructed"],
+        task_subtypes=["Sentiment/Hate speech"],
+        license="cc-by-4.0",
+        socioeconomic_status="high",
+        annotations_creators="expert-annotated",
+        dialect=None,
```
Comment (isaac-chung) [inline mteb/tasks/Classification/multilingual/ToxicChatClassification.py:69]:
```suggestion
        dialect=[],
```

File: mteb/tasks/Classification/multilingual/ToxicChatClassification.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        text_creation=None,
+        bibtex_citation=None,
```
Comment (isaac-chung) [inline mteb/tasks/Classification/multilingual/ToxicChatClassification.py:71]: Let's add the citation provided on the dataset page:
```
@misc{lin2023toxicchat,
      title={ToxicChat: Unveiling Hidden Challenges of Toxicity Detection in Real-World User-AI Conversation}, 
      author={Zi Lin and Zihan Wang and Yongqi Tong and Yangkun Wang and Yuxin Guo and Yujia Wang and Jingbo Shang},
      year={2023},
      eprint={2310.17389},
      archivePrefix={arXiv},
      primaryClass={cs.CL}
}

```

File: mteb/tasks/Classification/multilingual/ToxicChatClassification.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        main_score="accuracy",
+        date=("2023-10-26", "2024-01-31"),
+        form=["written"],
+        domains=["Constructed"],
+        task_subtypes=["Sentiment/Hate speech"],
+        license="cc-by-4.0",
+        socioeconomic_status="high",
+        annotations_creators="expert-annotated",
+        dialect=None,
+        text_creation=None,
```
Comment (isaac-chung) [inline mteb/tasks/Classification/multilingual/ToxicChatClassification.py:70]:
```suggestion
        text_creation="found",
```

File: mteb/tasks/Classification/multilingual/ToxicChatClassification.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        # only use human-annotated data
+        self.dataset = self.dataset["test"].filter(lambda x: x["human_annotation"])
+        keep_cols = ["user_input", "toxicity"]
+        rename_dict = dict(zip(keep_cols, ["text", "label"]))
+        remove_cols = [col for col in self.dataset.column_names if col not in keep_cols]
+        self.dataset = self.dataset.rename_columns(rename_dict)
+        self.dataset = self.dataset.remove_columns(remove_cols)
+        self.dataset = self.dataset.class_encode_column("label")
+        self.dataset = self.dataset.train_test_split(
+            test_size=0.5, seed=self.seed, stratify_by_column="label"
```
Comment (isaac-chung) [inline mteb/tasks/Classification/multilingual/ToxicChatClassification.py:86]: Great that it's stratified! Also, since test_size is 0.5, would `n_samples` be half of ~5k (length of test set) instead of 1427?

**Review by isaac-chung — COMMENTED** [review]
Thank you for your contribution! Just a few comments below.

Feel free to add points for yourself (6 for each new language in classification - you can do a quick search in the codebase on the lang code, and 2 for each of the rest) and 2 points for me for the review.

File: mteb/tasks/Classification/multilingual/ToxicChatClassification.py (hunk 4 above)
Comment (dokato) [reply mteb/tasks/Classification/multilingual/ToxicChatClassification.py:93]: Correct, except that I filtered only human annotated rows above.

**Comment by isaac-chung** [comment]
A task is like Classification, Retrieval etc. So this PR is extending language support for an existing task. 

For each of these languages, add 2 + 4 = 6 points as they are new languages in Classification. So that's 9x6=54
```
[
    "glg-Latn",
    "lit-Latn",
    "bre-Latn",
    "mlg-Latn",
    "que-Latn",
    "eus-Latn",
    "nor-Latn",
    "cat-Latn",
    "ido-Latn",
]
```

For each of the rest, add 2 points each. So that's 20x2 = 40.

Which makes the total = 94 pts for you, 2 pts for me. Great work!

File: mteb/tasks/Classification/multilingual/ToxicChatClassification.py (hunk 4 above)
Comment (isaac-chung) [reply mteb/tasks/Classification/multilingual/ToxicChatClassification.py:93]: Aha good call out. I missed the comment above. Thanks!

**Review by isaac-chung — APPROVED** [review]

**Comment by dokato** [comment]
Amazing, thank you @isaac-chung 🎉

**Comment by isaac-chung** [comment]
@imenelydiaker I should have caught this at review time. @dokato before I open a PR to fix points and move this under English, did you mean to add a different dataset that covers add these languages?

**Comment by dokato** [comment]
Sorry, as I'm new to the project I wasn't sure what is the definition of "multilingual" here. Since there were multiple languages in the text, I detected them all and listed as appearing in the set.

> did you mean to add a different dataset that covers add these languages?

Not sure what do you mean by that either? I don't have anything related to toxic chat classification for sure, but I started working on `Davlan/sib200` that clearly has many languages. I'll submit a PR for this soon.

**Comment by dokato** [comment]
BTW happy to help amend it if you instruct me what are best steps?

**Comment by imenelydiaker** [comment]
> Sorry, as I'm new to the project I wasn't sure what is the definition of "multilingual" here. Since there were multiple languages in the text, I detected them all and listed as appearing in the set.
> 

How did you detect them? The problem is that the paper says that 99% of these conversations are in English and the HF dataset card says it's an English dataset also. For multilingual datasets, we require one subset per language to be able to evaluate each language on its own. 
If this dataset contains mixed texts, which means a text with multiple languages in it, we'll take the more important language and the rest of it will be treated as OOV tokens by models I suppose. So in this case, it would be English.

**Comment by KennethEnevoldsen** [comment]
> we require one subset per language to be able to evaluate each language on its own.

We do have datasets where that is not the case (e.g. language classification).

**Comment by imenelydiaker** [comment]
> > we require one subset per language to be able to evaluate each language on its own.
> 
> We do have datasets where that is not the case (e.g. language classification).
> 
I think this [dataset](https://github.com/embeddings-benchmark/mteb/blob/main/mteb/tasks/Classification/multilingual/IndicLangClassification.py) considers each language as a label, right? 
Here, there's no indicator on the language of the samples, and the paper introducing it doesn't say that there is anything different from English.
The only problem I see when mixing languages, is that we deviate from the initial goal of evaluating a model for a task on a specific language. 
Although, having mixed-languages samples (one sample with multiple languages in the same text) is cool and multilingual models should be able to handle them, but I'm not sure it's right to evaluate a language that is mixed with another one, it becomes familiar and informal languages evaluation, no?

**Comment by dokato** [comment]
> How did you detect them?

I went through all texts and and used `langid.classify` to detect the language code.

**Comment by dokato** [comment]
So step to fix it following @imenelydiaker comments:

1) Move `multilingual/ToxicChatClassification.py` to `eng/ToxicChatClassification.py`
2) Remove _LANG and `eval_langs` becomes `eng-Latn`

Sth else?Should we do sth about other languages or as you said model will treat them as OOS and we can live with that?

**Comment by KennethEnevoldsen** [comment]
@imenelydiaker feel free to create an issue to discuss this further (I believe it is a relevant discussion, but not too related to this)

@dokato these seems like good changes I believe I would change the eval languages to eng-Latn but with a comment in the code that it also contains other languages (potentially in the description as well).

**Comment by isaac-chung** [comment]
Re: Fix, do compare to what's already been included in https://github.com/embeddings-benchmark/mteb/pull/527 by @imenelydiaker .

**Comment by imenelydiaker** [comment]
> So step to fix it following @imenelydiaker comments:
> 
> 1) Move `multilingual/ToxicChatClassification.py` to `eng/ToxicChatClassification.py`
> 2) Remove _LANG and `eval_langs` becomes `eng-Latn`
> 
> Sth else?Should we do sth about other languages or as you said model will treat them as OOS and we can live with that?

I've already addressed this in PR #527
